packages/calypso-e2e/src/lib/components/editor-component.ts

```diff
@@ -0,0 +1,99 @@
+import { Page, Locator } from 'playwright';
+
+const EDITOR_TIMEOUT = 60 * 1000;
+
+/**
+ * Represents the Editor component.
+ */
+export class EditorComponent {
+	private page: Page;
+	private parentLocator: Locator | null;
+	private canvasLocator: Locator | null;
+
+	/** */
+	constructor( page: Page ) {
+		this.page = page;
+		this.parentLocator = null;
+		this.canvasLocator = null;
+	}
+
+	/**
+	 * Returns the Editor parent locator. It will automatically resolve to the
+	 * proper locator, regardless if the Editor is Gutenframed or not.
+	 */
+	async parent(): Promise< Locator > {
+		if ( this.parentLocator ) {
+			return this.parentLocator;
+		}
+
+		try {
+			this.parentLocator = await Promise.any( [
+				this.waitForFramedEditor(),
+				this.waitForUnframedEditor(),
+			] );
+		} catch ( _error ) {
+			throw new Error( 'Timed out waiting for the Editor' );
+		}
+
+		return this.parentLocator;
+	}
+
+	/**
+	 * Returns the Editor canvas locator. It will automatically resolve to the
+	 * proper locator, regardless if the canvas is iframed or not.
+	 */
+	async canvas(): Promise< Locator > {
+		if ( this.canvasLocator ) {
+			return this.canvasLocator;
+		}
+
+		try {
+			this.canvasLocator = await Promise.any( [
+				this.waitForFramedCanvas(),
+				this.waitForUnframedCanvas(),
+			] );
+		} catch ( _error ) {
+			throw new Error( 'Timed out waiting for the Editor canvas' );
+		}
+
+		return this.canvasLocator;
+	}
+
+	/** */
```

Comment:
**issue:blocking** - probably just an oversight, but all new methods in this PR needs some form of JSDocs.

Reply:
https://github.com/Automattic/wp-calypso/pull/76085/commits/912231aadca2662b6527737f38860433075852e5